lib/HttpClient.js

```diff
@@ -150,8 +162,32 @@ function rawRequest(opts, cb) {
         ]);
     });
 
+    var handleRedirect = function (_err, _req, _res) {
+        if (_err ||
+            !opts.followRedirects ||
+            REDIRECT_CODES.indexOf(_res.statusCode) < 0) {
+            return false;
+        }
+
+        if (opts.maxRedirects &&
+            opts.redirects >= opts.maxRedirects) {
+            var err = new TooManyRedirectsError(opts.redirects);
+            _req.emit('result', err, _res);
+        } else {
+            opts.redirects = (opts.redirects || 0) + 1;
+            _res.forceGet = _res.statusCode !== 307 && _req.method !== 'HEAD';
```

Comment:
307 should not change the request method according to the https://tools.ietf.org/html/rfc7231#section-6.4.7 .

301/302/303 change the request method when following to the new location. 301/302 are not required to do this, but that's the way many http clients work, so I decided to make it consistent.